mentat/diff_context.py

```diff
@@ -91,6 +91,9 @@ def annotate_file_message(code_message: list[str], annotations: list[DiffAnnotat
 
 
 class DiffContext:
+    target: str = ""
```

Comment:
Does mentat use the diff for anything besides rendering annotations in the code message and putting information in the tree viewer? Should we totally delete this file and move the functionality to ragdaemon?

Also when I run `mentat --diff main` on this branch I get the following error:
![image](https://github.com/AbanteAI/mentat/assets/385534/d8f99ec6-4325-4d7a-90b6-13a40a3c0002)